hal/lib/Athena/Digital.cpp

```diff
@@ -21,59 +24,29 @@
 
 static_assert(sizeof(uint32_t) <= sizeof(void *), "This file shoves uint32_ts into pointers.");
 
-static const uint32_t kExpectedLoopTiming = 40;
-static const uint32_t kDigitalPins = 26;
-static const uint32_t kPwmPins = 20;
-static const uint32_t kRelayPins = 8;
-static const uint32_t kNumHeaders = 10; // Number of non-MXP pins
 
-/**
- * kDefaultPwmPeriod is in ms
- *
- * - 20ms periods (50 Hz) are the "safest" setting in that this works for all devices
- * - 20ms periods seem to be desirable for Vex Motors
- * - 20ms periods are the specified period for HS-322HD servos, but work reliably down
- *      to 10.0 ms; starting at about 8.5ms, the servo sometimes hums and get hot;
- *      by 5.0ms the hum is nearly continuous
- * - 10ms periods work well for Victor 884
- * - 5ms periods allows higher update rates for Luminary Micro Jaguar speed controllers.
- *      Due to the shipping firmware on the Jaguar, we can't run the update period less
- *      than 5.05 ms.
- *
- * kDefaultPwmPeriod is the 1x period (5.05 ms).  In hardware, the period scaling is implemented as an
- * output squelch to get longer periods for old devices.
- */
-static const float kDefaultPwmPeriod = 5.05;
-/**
- * kDefaultPwmCenter is the PWM range center in ms
- */
-static const float kDefaultPwmCenter = 1.5;
-/**
- * kDefaultPWMStepsDown is the number of PWM steps below the centerpoint
- */
-static const int32_t kDefaultPwmStepsDown = 1000;
-static const int32_t kPwmDisabled = 0;
 
 struct DigitalPort {
   Port port;
   uint32_t PWMGeneratorID;
 };
 
+
 // Create a mutex to protect changes to the digital output values
-static priority_recursive_mutex digitalDIOMutex;
+priority_recursive_mutex digitalDIOMutex;
```

Comment:
The extern definitions are defined in an internal header. The issue with these is SPI, I2C, PWM and DIO all require access to each other because of the MXP header. Before, all digital was in 1 file, however that file was going to get WAY too big, and even bigger with the resource change. Splitting the file seems like the best choice, but then to get cross access you either need to declare the variables extern or have functions to pass those pointers around, at which point either would be declared as part of the API. Both would be placed in an internal header,  so users by default would not have access to them, but if the files are split I cannot think of a way to not export either the extern variables or functions to grab those variables.